.github/workflows/check-catalyst.yaml

```diff
@@ -132,7 +132,7 @@ jobs:
     - name: Install Deps
       if: steps.cache-llvm-build.outputs.cache-hit != 'true'
       run: |
-        sudo apt-get install -y python3 python3-pip cmake ninja-build clang lld
+        sudo apt-get install -y python3 python3-pip cmake ninja-build clang lld ccache
```

Comment:
Why are we adding ccache for llvm and mhlo? We only backup the cache files for the quantum dialect build so I think it would be useless.

Reply:
The build process fails when we actually attempt to build with `ccache`. The alternative is to specify `COMPILER_LAUNCHER= ` to leave it empty. I will change it then.

Reply:
Yes I think we already use the `COKMPILER_LAUNCHER=""` pattern in several places in the action for exactly this purpose.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/143/commits/7f03eb4ef133ac3d3ee2c691192a892d40b4ff65, https://github.com/PennyLaneAI/catalyst/pull/143/commits/592e288d319e17c95b980d3fd8e0e57d529ae237,
https://github.com/PennyLaneAI/catalyst/pull/143/commits/7d02284d887dff83a284f72b7af445d85b52ebeb